internal/schemautil/schemautil.go

```diff
@@ -538,3 +542,15 @@ func Remarshal(in, out any) error {
 	d.UseNumber()
 	return d.Decode(out)
 }
+
+// MergeSchemas merges multiple schema maps into the base schema.
+// Later schemas take precedence over earlier ones in case of key conflicts.
+func MergeSchemas(base map[string]*schema.Schema, schemas ...map[string]*schema.Schema) map[string]*schema.Schema {
+	for _, s := range schemas {
+		for k, v := range s {
+			base[k] = v
+		}
+	}
+
+	return base
+}
```

Comment:
nit: It is more or less a generic operation, already covered in `lo`.
https://github.com/samber/lo?tab=readme-ov-file#assign